interface/core.xhr.json

```diff
@@ -0,0 +1,53 @@
+{
+  "name": "core.xhr",
+  "api": {
+    // @param {String} method - GET/POST/DELETE
+    // @param {String} url
+    // @param {Boolean} async
+    // @param {String} user
+    // @param {String} password
+    "open": {
+      "type": "method",
+      "value": [ "string", "string", "boolean", "string", "string" ],
+      "err": {
+        "errcode": "string",
+        "message": "string"
+      }
+    },
+    // Specify a single element in the argument object,
+    // depending on the type of data to send (e.g. in POST)
+    "send": { "type": "method", "value": [ { "string": "string", "buffer": "buffer" } ] },
+    "abort": { "type": "method", "value": [] },
+    // @param {String} header
+    // @return {String} value
+    "getResponseHeader": { "type": "method", "value": [ "string" ], "ret": "string" },
+    // @param {String} all headers in a String
+    "getAllResponseHeaders": { "type": "method", "value": [], "ret": "string" },
+    // @param {String} header
+    // @param {String} value
+    "setRequestHeader": { "type": "method", "value": [ "string", "string" ] },
+    // @param {String} mimetype
+    "overrideMimeType": { "type": "method", "value": [ "string" ] },
+    "getReadyState": { "type": "method", "value": [], "ret": "number" },
+    "getResponse": { "type": "method", "value": [], "ret": { "string": "string", "buffer": "buffer", "object": "object" } },
+    "getResponseText": { "type": "method", "value": [], "ret": "string" },
+    "getResponseURL": { "type": "method", "value": [], "ret": "string" },
+    "getResponseType": { "type": "method", "value": [], "ret": "string" },
+    "setResponseType": { "type": "method", "value": [  "string" ] },
+    "getStatus": { "type": "method", "value": [], "ret": "number" },
+    "getStatusText": { "type": "method", "value": [], "ret": "string" },
+    "getTimeout": { "type": "method", "value": [], "ret": "number" },
+    "setTimeout": { "type": "method", "value": [ "number" ] },
+    "getWithCredentials": { "type": "method", "value": [], "ret": "boolean" },
+    "setWithCredentials": { "type": "method", "value": [ "boolean" ] },
+
+    "onloadstart": { "type": "event", "value": { "lengthComputable": "boolean", "loaded": "number", "total": "number" } },
```

Comment:
I've never used it before, so I'm a little confused by the intended use. I
thought this was populated off window.XMLHttpRequestUpload, and not
window.XMLHttpRequest, so it would be a new core.xhru api. is that a
correct interpretation?

On Thu, Jan 8, 2015 at 11:28 AM, Benjamin M. Schwartz <
notifications@github.com> wrote:

> In interface/core.xhr.json
> <https://github.com/freedomjs/freedom/pull/188#discussion-diff-22661994>:
>
> > +    // @param {String} mimetype
> > +    "overrideMimeType": { "type": "method", "value": [ "string" ] },
> > +    "getReadyState": { "type": "method", "value": [], "ret": "number" },
> > +    "getResponse": { "type": "method", "value": [], "ret": { "string": "string", "buffer": "buffer", "object": "object" } },
> > +    "getResponseText": { "type": "method", "value": [], "ret": "string" },
> > +    "getResponseURL": { "type": "method", "value": [], "ret": "string" },
> > +    "getResponseType": { "type": "method", "value": [], "ret": "string" },
> > +    "setResponseType": { "type": "method", "value": [  "string" ] },
> > +    "getStatus": { "type": "method", "value": [], "ret": "number" },
> > +    "getStatusText": { "type": "method", "value": [], "ret": "string" },
> > +    "getTimeout": { "type": "method", "value": [], "ret": "number" },
> > +    "setTimeout": { "type": "method", "value": [ "number" ] },
> > +    "getWithCredentials": { "type": "method", "value": [], "ret": "boolean" },
> > +    "setWithCredentials": { "type": "method", "value": [ "boolean" ] },
> > +
> > +    "onloadstart": { "type": "event", "value": { "lengthComputable": "boolean", "loaded": "number", "total": "number" } },
>
> Have you looked at the upload listener
> <https://xhr.spec.whatwg.org/#xmlhttprequestupload> thing?
>
> It looks like its only purpose is to expose this same set of events, but
> in reference to the upload instead of the download. Maybe the best solution
> is to have a parallel set of events like "onuploadprogress",
> "onuploadloadend", etc.. That should be enough to allow a proper polyfill.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freedomjs/freedom/pull/188/files#r22661994>.
>